### PR TITLE
TextInput Instance is created

### PR DESCRIPTION
**Pull request by JOHNFLEURIMOND**

### CODE REVIEW

File: services-js/commissions-app/src/client/common/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        size={20}
+        placeholder={this.props.placeholder}
+        type=""
+        className="txt-f txt-f--sm"
+      />
+    );
+  }
+}
+
+export default class FormWithElement extends React.Component {
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.tsx]: You don’t need this component. Focus just on the input field piece.

File: services-js/commissions-app/src/client/common/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import Head from 'next/head';
+import ReachDOM from 'react-dom';
+
+export class TextInput extends React.Component {
+  render() {
+    return (
+      <input
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.tsx]: You’re missing the `<label>` element and other things.

Remember we talked about how this component should have the whole `<div className="txt">…</div>` piece.

**Review by fionawhim — CHANGES_REQUESTED** [review]

File: services-js/commissions-app/src/client/common/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import ReachDOM from 'react-dom';
+
+export default class FormWithElement extends React.Component {
+  render() {
+    return (
+      <div className="fs">
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.tsx]: Remember, we don’t want the form. We want a reusable text input _field_.

You had it right when you posted this:

image

You just needed to use `props` so it wasn’t hard-coded to “First Name”

Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: I thought by leaving (name=" " <-props?)  would do that. I do not know how to do that.

Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: Do you want me to create a constructor and have the state with all the props to handleUpdate? I am confused. You wanted a single textinput that could be used multiple times to create your form.

Comment (fionawhim) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: Can you get the code back to where it was when you posted that screenshot? Then we can work through an example of bringing in props.

Comment (fionawhim) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: Looks like you just need to remove the `"fs-c"` div and leave the contents.

Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: I removed that div and it did not do anything

File: services-js/commissions-app/src/client/common/TextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <div className="fs-c">
+          <div className="txt m-b300">
+            <label
+              htmlFor="FeedbackForm-${this.props.name}"
+              className="txt-l txt-l--sm"
+            >
+              {this.props.name}
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.tsx]: Remember that `name` is a form `<input>` name, so it’ll be something like `firstName`. You’ll want a different prop for the label.

File: services-js/commissions-app/src/client/common/TextInput.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                  {' '}
+                  Required
+                </span>
+              )}
+            </label>
+            <input
+              name={this.props.name}
+              className="txt-f txt-f--sm"
+              placeholder={this.props.placeholder}
+              id=""
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.tsx]: Need to fill this in to match `htmlFor` in the `<label>` to associate the two together.

File: services-js/commissions-app/src/pages/commissions/apply.tsx [hunk 6]
```diff
@@ -24,7 +24,7 @@ export default class IndexPage extends React.Component<Props, State> {
   state: State = {
     emailMatch: true,
   };
-  //
+  ///
```
Comment (fionawhim) [inline services-js/commissions-app/src/pages/commissions/apply.tsx]: Remove this comment.

File: services-js/commissions-app/src/client/common/TextInput.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import TextInput from './TextInput';
+
+storiesOf('TextInput', module).add('default', () => <TextInput />);
```
Comment (fionawhim) [inline services-js/commissions-app/src/client/common/TextInput.stories.tsx]: Update this to send props to `<TextInput />`

You should also add a story to show off the “required” version.

See docs for Storybook: https://storybook.js.org/basics/writing-stories/

(Really this means chaining another call to `add` off of the one that’s already there.)

File: services-js/commissions-app/src/pages/commissions/apply.tsx [hunk 8]
```diff
@@ -55,249 +55,51 @@ export default class IndexPage extends React.Component<Props, State> {
...
-                      type="text"
-                      className="txt-f txt-f--sm"
-                      size={10}
-                      placeholder="Confirm Email"
-                      onBlur={this.handleBlur}
-                    />
-                  </div>
+                  <TextInput
+                    name="firstName"
+                    placeholder="First Name "
```
Comment (fionawhim) [inline services-js/commissions-app/src/pages/commissions/apply.tsx]: Remove trailing space from `placeholder`, here and below.

**Review by fionawhim — CHANGES_REQUESTED** [review]

File: services-js/commissions-app/src/client/common/TextInput.tsx (hunk 4 above)
Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: So something like this?         --->      htmlFor="FeedbackForm-${this.props.firstName}"

Comment (fionawhim) [reply services-js/commissions-app/src/client/common/TextInput.tsx]: This is about the text within the `<label>`. The `htmlFor` is totally fine as it is.

File: services-js/commissions-app/src/client/common/TextInput.stories.tsx (hunk 7 above)
Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.stories.tsx]: TextInput.stories.tsx is already created so when you ask to add a story to show off required. Storybook is already showing the required <span>

Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.stories.tsx]: <span>

Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/client/common/TextInput.stories.tsx]: span

**Review by fionawhim — APPROVED** [review]

**Comment by fionawhim** [comment]
@JOHNFLEURIMOND The code looks good, but don’t merge it just yet.

If you look at the PR, the “Percy” check has failed because Percy found UI changes in this PR.

To resolve this, click the "Details" link text to Percy and take a look at the visual changes. If they seem good, click the green checkmark in Percy and come back here and merge.

**Pull request merged**